akka-stream/src/main/scala/DocumentStage.scala

```diff
@@ -43,12 +43,10 @@ private[akkastream] class DocumentStage[T](
       @inline private def tailable = cursor.wrappee.tailable
 
       private var request: () => Future[Option[Response]] = { () =>
+        // Call onFirst immediately to make sure only one request is sent.
         cursor.makeRequest(maxDocs).andThen {
-          case Success(r) if (
-            tailable && r.reply.numberReturned > 0
-          ) => onFirst()
-
-          case Success(_) if (!tailable) => onFirst()
+          case Success(_) =>
```

Comment:
But it did happen. This is evidence from the mongodb log:
```
2017-08-10T00:48:01.405+0800 I COMMAND  [conn2948] query local.oplog.rs query: { $readPreference: { mode: "primary" }, $query: { ns: { $in: [ "play-community.common-doc", "play-community.common-article", "play-community.common-qa" ] }, ts: { $gte: Timestamp 1502282626000|1 } } } planSummary: COLLSCAN cursorid:27724224577 ntoreturn:2147483647 ntoskip:0 keysExamined:0 docsExamined:229305 numYields:1792 nreturned:0 reslen:20 locks:{ Global: { acquireCount: { r: 3586 } }, Database: { acquireCount: { r: 1793 } }, oplog: { acquireCount: { r: 1793 } } } 288ms
2017-08-10T00:48:01.996+0800 I COMMAND  [conn2952] query local.oplog.rs query: { $readPreference: { mode: "primary" }, $query: { ns: { $in: [ "play-community.common-doc", "play-community.common-article", "play-community.common-qa" ] }, ts: { $gte: Timestamp 1502282626000|1 } } } planSummary: COLLSCAN cursorid:27865157825 ntoreturn:2147483647 ntoskip:0 keysExamined:0 docsExamined:229305 numYields:1792 nreturned:0 reslen:20 locks:{ Global: { acquireCount: { r: 3586 } }, Database: { acquireCount: { r: 1793 } }, oplog: { acquireCount: { r: 1793 } } } 174ms
2017-08-10T00:48:02.490+0800 I COMMAND  [conn2953] query local.oplog.rs query: { $readPreference: { mode: "primary" }, $query: { ns: { $in: [ "play-community.common-doc", "play-community.common-article", "play-community.common-qa" ] }, ts: { $gte: Timestamp 1502282626000|1 } } } planSummary: COLLSCAN cursorid:26267358311 ntoreturn:2147483647 ntoskip:0 keysExamined:0 docsExamined:229305 numYields:1792 nreturned:0 reslen:20 locks:{ Global: { acquireCount: { r: 3586 } }, Database: { acquireCount: { r: 1793 } }, oplog: { acquireCount: { r: 1793 } } } 143ms
2017-08-10T00:48:03.108+0800 I COMMAND  [conn2948] query local.oplog.rs query: { $readPreference: { mode: "primary" }, $query: { ns: { $in: [ "play-community.common-doc", "play-community.common-article", "play-community.common-qa" ] }, ts: { $gte: Timestamp 1502282626000|1 } } } planSummary: COLLSCAN cursorid:28104903696 ntoreturn:2147483647 ntoskip:0 keysExamined:0 docsExamined:229305 numYields:1792 nreturned:0 reslen:20 locks:{ Global: { acquireCount: { r: 3586 } }, Database: { acquireCount: { r: 1793 } }, oplog: { acquireCount: { r: 1793 } } } 115ms
2017-08-10T00:48:04.006+0800 I COMMAND  [conn2946] query local.oplog.rs query: { $readPreference: { mode: "primary" }, $query: { ns: { $in: [ "play-community.common-doc", "play-community.common-article", "play-community.common-qa" ] }, ts: { $gte: Timestamp 1502282626000|1 } } } planSummary: COLLSCAN cursorid:26358384401 ntoreturn:2147483647 ntoskip:0 keysExamined:0 docsExamined:229305 numYields:1791 nreturned:0 reslen:20 locks:{ Global: { acquireCount: { r: 3584 } }, Database: { acquireCount: { r: 1792 } }, oplog: { acquireCount: { r: 1792 } } } 102ms
2017-08-10T00:48:05.694+0800 I COMMAND  [conn2947] query local.oplog.rs query: { $readPreference: { mode: "primary" }, $query: { ns: { $in: [ "play-community.common-doc", "play-community.common-article", "play-community.common-qa" ] }, ts: { $gte: Timestamp 1502282626000|1 } } } planSummary: COLLSCAN cursorid:27968011900 ntoreturn:2147483647 ntoskip:0 keysExamined:0 docsExamined:229305 numYields:1792 nreturned:0 reslen:20 locks:{ Global: { acquireCount: { r: 3586 } }, Database: { acquireCount: { r: 1793 } }, oplog: { acquireCount: { r: 1793 } } } 105ms
2017-08-10T00:48:05.837+0800 I COMMAND  [conn2953] query local.oplog.rs query: { $readPreference: { mode: "primary" }, $query: { ns: { $in: [ "play-community.common-doc", "play-community.common-article", "play-community.common-qa" ] }, ts: { $gte: Timestamp 1502282626000|1 } } } planSummary: COLLSCAN cursorid:27466019936 ntoreturn:2147483647 ntoskip:0 keysExamined:0 docsExamined:229305 numYields:1792 nreturned:0 reslen:20 locks:{ Global: { acquireCount: { r: 3586 } }, Database: { acquireCount: { r: 1793 } }, oplog: { acquireCount: { r: 1793 } } } 103ms
2017-08-10T00:48:07.295+0800 I COMMAND  [conn2944] query local.oplog.rs query: { $readPreference: { mode: "primary" }, $query: { ns: { $in: [ "play-community.common-doc", "play-community.common-article", "play-community.common-qa" ] }, ts: { $gte: Timestamp 1502282626000|1 } } } planSummary: COLLSCAN cursorid:29004186793 ntoreturn:2147483647 ntoskip:0 keysExamined:0 docsExamined:229305 numYields:1794 nreturned:0 reslen:20 locks:{ Global: { acquireCount: { r: 3590 } }, Database: { acquireCount: { r: 1795 } }, oplog: { acquireCount: { r: 1795 } } } 189ms
2017-08-10T00:48:09.208+0800 I COMMAND  [conn2949] query local.oplog.rs query: { $readPreference: { mode: "primary" }, $query: { ns: { $in: [ "play-community.common-doc", "play-community.common-article", "play-community.common-qa" ] }, ts: { $gte: Timestamp 1502282626000|1 } } } planSummary: COLLSCAN cursorid:27747020331 ntoreturn:2147483647 ntoskip:0 keysExamined:0 docsExamined:229305 numYields:1792 nreturned:0 reslen:20 locks:{ Global: { acquireCount: { r: 3586 } }, Database: { acquireCount: { r: 1793 } }, oplog: { acquireCount: { r: 1793 } } } 182ms
2017-08-10T00:48:09.995+0800 I COMMAND  [conn2946] query local.oplog.rs query: { $readPreference: { mode: "primary" }, $query: { ns: { $in: [ "play-community.common-doc", "play-community.common-article", "play-community.common-qa" ] }, ts: { $gte: Timestamp 1502282626000|1 } } } planSummary: COLLSCAN cursorid:27832447502 ntoreturn:2147483647 ntoskip:0 keysExamined:0 docsExamined:229306 numYields:1792 nreturned:0 reslen:20 locks:{ Global: { acquireCount: { r: 3586 } }, Database: { acquireCount: { r: 1793 } }, oplog: { acquireCount: { r: 1793 } } } 180ms
2017-08-10T00:48:11.977+0800 I COMMAND  [conn2952] query local.oplog.rs query: { $readPreference: { mode: "primary" }, $query: { ns: { $in: [ "play-community.common-doc", "play-community.common-article", "play-community.common-qa" ] }, ts: { $gte: Timestamp 1502282626000|1 } } } planSummary: COLLSCAN cursorid:30008643934 ntoreturn:2147483647 ntoskip:0 keysExamined:0 docsExamined:229306 numYields:1792 nreturned:0 reslen:20 locks:{ Global: { acquireCount: { r: 3586 } }, Database: { acquireCount: { r: 1793 } }, oplog: { acquireCount: { r: 1793 } } } 176ms
2017-08-10T00:48:13.824+0800 I COMMAND  [conn2946] query local.oplog.rs query: { $readPreference: { mode: "primary" }, $query: { ns: { $in: [ "play-community.common-doc", "play-community.common-article", "play-community.common-qa" ] }, ts: { $gte: Timestamp 1502282626000|1 } } } planSummary: COLLSCAN cursorid:28571059530 ntoreturn:2147483647 ntoskip:0 keysExamined:0 docsExamined:229306 numYields:1791 nreturned:0 reslen:20 locks:{ Global: { acquireCount: { r: 3584 } }, Database: { acquireCount: { r: 1792 } }, oplog: { acquireCount: { r: 1792 } } } 136ms
2017-08-10T00:48:13.837+0800 F -        [rsSync] out of memory.

 0x55b742b40911 0x55b742b3ff44 0x55b742cfd54b 0x55b743638b34 0x55b74250d9c1 0x55b74262f0ce 0x55b74260ed9b 0x55b7435b6fc0 0x7fe5f16f4184 0x7fe5f1420ffd
----- BEGIN BACKTRACE -----
{"backtrace":[{"b":"55B7415CF000","o":"1571911","s":"_ZN5mongo15printStackTraceERSo"},{"b":"55B7415CF000","o":"1570F44","s":"_ZN5mongo29reportOutOfMemoryErrorAndExitEv"},{"b":"55B7415CF000","o":"172E54B"},{"b":"55B7415CF000","o":"2069B34","s":"tc_new"},{"b":"55B7415CF000","o":"F3E9C1","s":"_ZNSt6vectorIN5mongo4repl10OplogEntryESaIS2_EE7reserveEm"},{"b":"55B7415CF000","o":"10600CE","s":"_ZN5mongo4repl8SyncTail16oplogApplicationEPNS0_22ReplicationCoordinatorE"},{"b":"55B7415CF000","o":"103FD9B","s":"_ZN5mongo4repl10RSDataSync4_runEv"},{"b":"55B7415CF000","o":"1FE7FC0"},{"b":"7FE5F16EC000","o":"8184"},{"b":"7FE5F1323000","o":"FDFFD","s":"clone"}],"processInfo":{ "mongodbVersion" : "3.4.6", "gitVersion" : "c55eb86ef46ee7aede3b1e2a5d184a7df4bfb5b5", "compiledModules" : [], "uname" : { "sysname" : "Linux", "release" : "4.4.0-81-generic", "version" : "#104~14.04.1-Ubuntu SMP Wed Jun 14 12:45:52 UTC 2017", "machine" : "x86_64" }, "somap" : [ { "b" : "55B7415CF000", "elfType" : 3, "buildId" : "C25286FEAB1A51FD1C0C00605BFA6244C141CC06" }, { "b" : "7FFF1DBE3000", "elfType" : 3, "buildId" : "19539EACE2FA4EFE68CD76233FD3335C3A7A1CB9" }, { "b" : "7FE5F260E000", "path" : "/lib/x86_64-linux-gnu/libssl.so.1.0.0", "elfType" : 3, "buildId" : "48A664AE6B0B4918A3EB0156C6364C4F084232FD" }, { "b" : "7FE5F2232000", "path" : "/lib/x86_64-linux-gnu/libcrypto.so.1.0.0", "elfType" : 3, "buildId" : "6B8997EA892A7FF37AC8CAA8F239D595251889BB" }, { "b" : "7FE5F202A000", "path" : "/lib/x86_64-linux-gnu/librt.so.1", "elfType" : 3, "buildId" : "AC72654C6338205F30190061C0D781CB0039B793" }, { "b" : "7FE5F1E26000", "path" : "/lib/x86_64-linux-gnu/libdl.so.2", "elfType" : 3, "buildId" : "EED41ABB999C74882F001C53979CC820ED15BA82" }, { "b" : "7FE5F1B20000", "path" : "/lib/x86_64-linux-gnu/libm.so.6", "elfType" : 3, "buildId" : "8F0318B9CC6FD523C2587A15C5447ABBB8CD813D" }, { "b" : "7FE5F190A000", "path" : "/lib/x86_64-linux-gnu/libgcc_s.so.1", "elfType" : 3, "buildId" : "36311B4457710AE5578C4BF00791DED7359DBB92" }, { "b" : "7FE5F16EC000", "path" : "/lib/x86_64-linux-gnu/libpthread.so.0", "elfType" : 3, "buildId" : "F48E96A1F4A549776CA4167095AD7527720D4B0E" }, { "b" : "7FE5F1323000", "path" : "/lib/x86_64-linux-gnu/libc.so.6", "elfType" : 3, "buildId" : "3217CA3A53A930C7BB1E5C83789D09B30B0F3B39" }, { "b" : "7FE5F286D000", "path" : "/lib64/ld-linux-x86-64.so.2", "elfType" : 3, "buildId" : "37AFDBB933B8409476E845DF5FB11BC77CBCEEE6" } ] }}
 mongod(_ZN5mongo15printStackTraceERSo+0x41) [0x55b742b40911]
 mongod(_ZN5mongo29reportOutOfMemoryErrorAndExitEv+0x84) [0x55b742b3ff44]
 mongod(+0x172E54B) [0x55b742cfd54b]
 mongod(tc_new+0x224) [0x55b743638b34]
 mongod(_ZNSt6vectorIN5mongo4repl10OplogEntryESaIS2_EE7reserveEm+0x91) [0x55b74250d9c1]
 mongod(_ZN5mongo4repl8SyncTail16oplogApplicationEPNS0_22ReplicationCoordinatorE+0x3EE) [0x55b74262f0ce]
 mongod(_ZN5mongo4repl10RSDataSync4_runEv+0x22B) [0x55b74260ed9b]
 mongod(+0x1FE7FC0) [0x55b7435b6fc0]
 libpthread.so.0(+0x8184) [0x7fe5f16f4184]
 libc.so.6(clone+0x6D) [0x7fe5f1420ffd]
-----  END BACKTRACE  -----
```
The total memory of the server is 2G. After running the tailing codes, the memory rose from  200m to 1.9G in 30 minutes from the output of htop command.  And the output from mongostat showed that the tailing codes sent  about 300 querying requests per second. Finally, mongod exhausted all memories and exited.